--- a/nginx-sticky/ngx_http_sticky_misc.c    2015-08-06 12:43:01.000000000 +0200
+++ b/nginx-sticky/ngx_http_sticky_misc.c    2017-10-06 11:48:16.399988930 +0200
@@ -9,6 +9,12 @@
 #include <ngx_http.h>
 #include <ngx_md5.h>
 #include <ngx_sha1.h>
+#ifndef MD5_DIGEST_LENGTH
+#include <openssl/md5.h>
+#endif
+#ifndef SHA_DIGEST_LENGTH
+#include <openssl/sha.h>
+#endif

 #include "ngx_http_sticky_misc.h"
